tests/admin_changelist/test_date_hierarchy.py

```diff
@@ -95,3 +96,11 @@ def test_invalid_params(self):
                 self.assertRaises(IncorrectLookupParameters),
             ):
                 self.assertDateParams(invalid_query, None, None)
+
+    @override_settings(ROOT_URLCONF="admin_changelist.urls")
+    def test_label_in_hierarchy(self):
+        self.client.force_login(self.superuser)
+        Event.objects.create(date=datetime(2017, 1, 1))
+        response = self.client.get(reverse("admin:admin_changelist_event_changelist"))
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, "Filter by")
```

Comment:
I think instead of testing the template we could test that the view provided the dynamic content.